src/libsyntax/parse/mod.rs

```diff
@@ -148,7 +148,7 @@ pub fn parse_stmt_from_source_str(name: String,
         name,
         source
     );
-    maybe_aborted(p.parse_stmt(), p)
+    maybe_aborted(panictry!(p.parse_stmt_nopanic()), p)
```

Comment:
In a lot of places you're changing to using nopanic versions of parse functions but wrapping them in panictry - what is the point of that?